### PR TITLE
enhancement: configurable share role icons

### PR DESCRIPTION
**Pull request by fschade**
## Description
Introduce themeable sharing role icons which are used in the sidebar sharing dialog.

## Related Issue
- Fixes https://github.com/owncloud/web/pull/10801
- Fixes https://github.com/owncloud/web/issues/10777

## Motivation and Context
give the individual user the ability to customize the look of the sharing role dropdown

## How Has This Been Tested?

- missing unit tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: packages/web-runtime/themes/owncloud/theme.json [hunk 1]
```diff
@@ -7,6 +7,40 @@
...
+        "label": "UnifiedRoleSpaceEditor",
+        "iconName": "pencil"
+      },
+      "312c0871-5ef7-4b3a-85b6-0e4074c64049": {
+        "label": "UnifiedRoleManager",
+        "iconName": "user-star"
+      },
+      "1c996275-f1c9-4e71-abdf-a42f6495e960": {
+        "label": "UnifiedRoleUploader",
+        "iconName": "pencil"
```
Comment (JammingBen) [inline packages/web-runtime/themes/owncloud/theme.json:38]: I know I suggested `pencil` earlier, but the upload icon makes more sense if I think about it.

```suggestion
        "iconName": "upload"
```

File: packages/web-runtime/themes/owncloud/theme.json [hunk 2]
```diff
@@ -7,6 +7,40 @@
       "accessDeniedHelp": "",
       "imprint": "",
       "privacy": ""
+    },
+    "roles": {
```
Comment (JammingBen) [inline packages/web-runtime/themes/owncloud/theme.json:11]: Just thinking a bit about the naming, maybe `shareRoles` would be more fitting 🤔 What do you think?

File: docs/theming/_index.md [hunk 3]
```diff
@@ -60,7 +60,8 @@ The structure of a valid `common` section is outlined below:
     "accessDeniedHelp": "",
     "imprint": "",
     "privacy": ""
-  }
+  }, 
+  "roles": {}
```
Comment (JammingBen) [inline docs/theming/_index.md:64]:
```suggestion
  "shareRoles": {}
```

File: docs/theming/_index.md [hunk 4]
```diff
@@ -387,6 +388,40 @@ A full template for your custom theme is provided below, and you can use the ins
       "accessDeniedHelp": "",
       "imprint": "",
       "privacy": ""
+    },
+    "roles": {
```
Comment (JammingBen) [inline docs/theming/_index.md:392]:
```suggestion
    "shareRoles": {
```

**Review by JammingBen — COMMENTED** [review]
Docs still need the naming adjustment, rest LGTM 👍

**Review by JammingBen — APPROVED** [review]

**Comment by kulmann** [comment]
This PR has caused a schema validation error on the default theme:

```js
ZodError: [
  {
    "code": "invalid_type",
    "expected": "object",
    "received": "undefined",
    "path": [
      "common",
      "shareRoles"
    ],
    "message": "Required"
  }
]
```

**Comment by JammingBen** [comment]
> This PR has caused a schema validation error on the default theme:

Did you run `pnpm build`? I can't reproduce it 🤔

**Comment by kulmann** [comment]
> > This PR has caused a schema validation error on the default theme:
> 
> Did you run `pnpm build`? I can't reproduce it 🤔

thank you 😆 🤦🏻‍♂️

**Pull request merged**